Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -244,5 +265,11 @@ extension CustomerIOImplementation {
                                     blockingGroups: [
                                         .identifiedProfile(identifier: currentlyIdentifiedProfileIdentifier)
                                     ])
+
+        if type == .screen {
+            hooks.screenViewHooks.forEach { hook in
+                hook.screenViewed(name: name)
+            }
+        }
```

Comment:
This is the core logic behind the PR. Moving this code here from the `automaticScreenView()` function.